File: 415-AddStrings/addStrings.java
public class Solution {
    public String addStrings(String num1, String num2) {
        if (num1.length() < num2.length() ) return addStrings(num2, num1);
        
        char[] arr1 = num1.toCharArray();
        char[] arr2 = num2.toCharArray();
        int carry = 0;
        
        for (int i = 0; i< arr1.length; i++) {
            if (i < arr2.length) {
                int sum = (arr1[arr1.length -1-i] - '0') + (arr2[arr2.length -1-i] - '0') + carry;
                carry = sum / 10;
                arr1[arr1.length -1-i] = (char) ('0' + sum % 10);
            } else {
                int sum = (arr1[arr1.length -1-i] - '0') + carry;
                carry = sum / 10;
                arr1[arr1.length -1 -i] = (char) ('0' + sum % 10);
            }
        }
        String rslt = new String(arr1);
        return carry == 1? ('1'+rslt) : rslt;
        
        
    }
}

File: 415-AddStrings/addStrings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        string res;
        if(num1.size()<num2.size()) return addStrings(num2,num1);
        int carry=0,i=num1.size()-1;
        for(int j=num2.size()-1;j>=0;j--){
            int sum=num1[i]-'0'+num2[j]-'0'+carry;
            char digit='0'+sum%10;
            res=digit+res;
            carry=sum>=10?1:0;
            i--;
        }

        while(i>=0){
            int sum=num1[i]-'0'+carry;
            char digit='0'+sum%10;
            res=digit+res;
            carry=sum>=10?1:0;
            i--;
        }
        return carry?"1"+res:res;
    }
};